packages/block-library/src/button/edit.js

```diff
@@ -72,6 +88,85 @@ function BorderPanel( { borderRadius = '', setAttributes } ) {
 	);
 }
 
+const handleLinkControlOnKeyDown = ( event ) => {
+	const { keyCode } = event;
+
+	if ( [ LEFT, DOWN, RIGHT, UP, BACKSPACE, ENTER ].indexOf( keyCode ) > -1 ) {
+		// Stop the key event from propagating up to ObserveTyping.startTypingInTextField.
+		event.stopPropagation();
+	}
+};
+
+const handleLinkControlOnKeyPress = ( event ) => {
+	event.stopPropagation();
+};
+
+function URLPicker( { isSelected, url, title, setAttributes, opensInNewTab, onToggleOpenInNewTab } ) {
+	const [ isURLPickerOpen, setIsURLPickerOpen ] = useState( false );
+	useEffect(
+		() => {
+			if ( ! isSelected ) {
+				setIsURLPickerOpen( false );
+			}
+		},
+		[ isSelected ]
+	);
+	const openLinkControl = () => {
+		setIsURLPickerOpen( true );
+	};
+	const linkControl = isURLPickerOpen && (
+		<LinkControl
+			className="wp-block-navigation-link__inline-link-input"
+			onKeyDown={ handleLinkControlOnKeyDown }
+			onKeyPress={ handleLinkControlOnKeyPress }
+			currentLink={ ! url && ! title ? null : { url, title } }
+			onLinkChange={ ( { title: newTitle = '', url: newURL = '' } ) => {
+				setAttributes( {
+					title: escape( newTitle ),
+					url: newURL,
+				} );
+			} }
+			currentSettings={ [
+				{
+					id: 'opensInNewTab',
+					title: __( 'Open in new tab' ),
+					checked: opensInNewTab,
+				},
+			] }
+			onSettingsChange={ ( setting, value ) => {
+				if ( setting === 'opensInNewTab' ) {
+					onToggleOpenInNewTab( value );
+				}
+			} }
```

Comment:
The  API is a bit weird to me. That's feedback that is not meant directly to this  PR but more about the LinkControl component. cc @getdave 
I think ideally,  the current link prop is just another setting and we should have a single onChange handler. Any particulra reason to separate both?
I also feel we should separate the definition of the settings (id, title) from the value (checked). 
Should we also absorb `handleLinkControlOnKeyDown` inside the component itself?

Reply:
I took a stab at this here https://github.com/WordPress/gutenberg/pull/19396